### PR TITLE
fix(content): Make sure components do not throw if some data is missing

### PR DESCRIPTION
**Pull request by k88hudson**
The reasoning behind this is that sometimes we get weirdly formatted history items, which ideally we will filter out, but at least we don't want to crash the whole parent component.

### CODE REVIEW

File: content-test/components/ActivityFeed.test.js [hunk 1]
```diff
@@ -37,6 +37,12 @@ describe("ActivityFeedItem", function() {
     el = ReactDOM.findDOMNode(instance);
   });
 
+  it("should not throw if on missing props", () => {
```
Comment (pdehaan) [inline content-test/components/ActivityFeed.test.js]: nit: not sure if you want "should not throw **if** missing props" or "should not throw **on** missing props".
Having both "**if on**" sounds a bit awkward (same w/ component-test/components/{SiteIcon,TopSites}.test.js below too)

Comment (k88hudson) [inline content-test/components/ActivityFeed.test.js]: oops brain fail

**Comment by pdehaan** [comment]
I think this needs a rebase to pick up the latest FF 45/Beta fixes.

**Comment by k88hudson** [comment]
@pdehaan I just realized karma is going to be a problem if you have stable firefox installed, since it will point to stable firefox

**Comment by k88hudson** [comment]
@pdehaan R? I added nightly back in for local test running so they don't fail.. as for the beta path, i'm going to file a ticket about it

File: yamscripts.yml [hunk 2]
```diff
@@ -41,7 +41,7 @@ scripts:
     # test:jpm: Run jpm tests
     jpm: jpm test -b beta
     # test:karma: Run content tests only
-    karma: karma start
+    karma: karma start --browsers FirefoxNightly
```
Comment (pdehaan) [inline yamscripts.yml]: So this is just overriding the `browsers` set in the [karma.conf.js:7](https://github.com/mozilla/activity-streams/blob/fix-empty-case/karma.conf.js#L7)?

**Comment by pdehaan** [comment]
I'm getting this locally:
```
SUMMARY:
✔ 64 tests completed
✖ 1 test failed

FAILED TESTS:
  ActivityFeedItem
    ✖ should not throw if on missing props
      Firefox 47.0.0 (Mac OS X 10.11.0)
    expected [Function] to not throw an error but 'TypeError: url is undefined' was thrown
    AssertionError@/Users/pdehaan/dev/github/activity-streams/content-test/index.js:354:19 <- webpack:///~/assertion-error/index.js:73:0
```

**Comment by pdehaan** [comment]
Travis-CI is passing, but I'm seeing the following warnings in the log:

> - ERROR: 'Warning: Failed propType: Required prop `url` was not specified in `ActivityFeedItem`.' ([source](https://travis-ci.org/mozilla/activity-streams/builds/110851472#L657))
>
> - ERROR: 'Warning: Failed propType: Required prop `sites[0].url` was not specified in `TopSites`.' ([source](https://travis-ci.org/mozilla/activity-streams/builds/110851472#L709))
> 
> - ERROR: 'Warning: Each child in an array or iterator should have a unique "key" prop. Check the render method of `TopSites`. See https://fb.me/react-warning-keys for more information.' ([source](https://travis-ci.org/mozilla/activity-streams/builds/110851472#L710))

**Comment by k88hudson** [comment]
Those warnings are actually expected, since we're making sure the components don't throw (just show errors)

**Comment by k88hudson** [comment]
@rlr what do you think?

**Comment by oyiptong** [comment]
Exceptions, even when expected, cause log entries to occur, at least in my experience. It be because we set the log to verbose mode.

**Comment by oyiptong** [comment]
As far as I can see, the tests actually not throwing anything, so I don't know why the errors are showing up

**Comment by rlr** [comment]
I see. Should we make those props have `isRequired=False` since we gracefully handle their absence? Or do we want the warnings in the console? I kind of lean towards just making it all optional since that's how we are handling it.

**Comment by k88hudson** [comment]
@rlr @oyiptong They are just validation errors from React; I think we do want to keep them, because it means there is something screwed up in our data source i.e. we're not filtering something properly and we probably want to know about it. However, I figured it's better just to warn rather than throw, which will destroy the whole component

**Comment by rlr** [comment]
sounds good. definitely warn over throw, especially on the released version :+1:

**Comment by oyiptong** [comment]
:+1:

**Pull request merged**